packages/amplify-graphql-function-transformer/src/graphql-function-transformer.ts

```diff
@@ -0,0 +1,174 @@
+import { DirectiveWrapper, MappingTemplate, TransformerPluginBase } from '@aws-amplify/graphql-transformer-core';
+import { TransformerContextProvider, TransformerSchemaVisitStepContextProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+import { obj, str, ref, printBlock, compoundExpression, qref, raw, iff } from 'graphql-mapping-template';
+import { FunctionResourceIDs, ResolverResourceIDs, ResourceConstants } from 'graphql-transformer-common';
+import { DirectiveNode, ObjectTypeDefinitionNode, InterfaceTypeDefinitionNode, FieldDefinitionNode } from 'graphql';
+
+type FunctionDirectiveConfiguration = {
+  name: string;
+  region: string | undefined;
+  resolverTypeName: string;
+  resolverFieldName: string;
+};
+
+const FUNCTION_DIRECTIVE_STACK = 'FunctionDirectiveStack';
+const directiveDefinition = /* GraphQL */ `
+  directive @function(name: String!, region: String) repeatable on FIELD_DEFINITION
+`;
+
+export class FunctionTransformer extends TransformerPluginBase {
```

Comment:
Unless I'm missing something, I don't think that's currently possible. Switching this class to extend `TransformerModelBase` generates 16 errors due to abstract classes not being implemented. Looking at the unimplemented methods, they don't seem to make sense in the context of `@function`.

Reply:
This kind of makes sense as our current implementation of `@function` is pretty limited. `@function` does not have any notion of query or mutation and does not provide anyway to declare a custom model where the query and mutation gets handled by function resolver. Also we don't have anyway where `@auth` would work on this.

Longer term we should look at how we can extend lamdba based models which could support everything that is supported in `@model`. It could be the same directive or may be a different directive.